Manak_Leah_ndvi_2022.py

```diff
@@ -0,0 +1,670 @@
+            # Apply cloud mask 
+            cloud_free_band = clean_band.where(
+                ~cropped_layer.isin(all_masked_values))
+            # Append list with the cloud free bands
+            site_bands.append(cloud_free_band)
+
+        # Calculate mean NDVI
+        mean_ndvi = np.nanmean(
+            (site_bands[1]-site_bands[0]) / (site_bands[1]+site_bands[0]))
+        # Append the mean NDVI to the empty site_info list
+        site_info.append(mean_ndvi)
+        # Append this list of lists the empty location_info list
+        location_info.append(site_info)
+
+# Create a pandas dataframe
+location_info_df = pd.DataFrame(location_info, columns=[
+                            "site", "date", "mean_ndvi"])
+
+# Set index on date
+indexed_location_df = location_info_df.set_index("date")
+
+final_NDVI_df = indexed_location_df.sort_values(by="date")
+
+# Call dataframe
+final_NDVI_df
+
+
+# In[9]:
+
+
+# Last sanity check before creating your plot (10 points)
+
+# Ensure that you call your dataframe at the bottom of the cell above
+# and that it has columns called: mean_ndvi and site
+
+# Ensure the data is stored in a dataframe.
+student_ndvi_df = _
+
+df_points = 0
+
+if isinstance(student_ndvi_df, pd.DataFrame):
+    print('\u2705 Your data is stored in a DataFrame!')
+    df_points +=2
+else:
+    print('\u274C It appears your data is not stored in a DataFrame. ',
+          'To see what type of object your data is stored in, check its type with type(object)')
+
+# Check that dataframe contains the appropriate number of NAN values
+if student_ndvi_df.mean_ndvi.isna().sum() == 15:
+    print('\u2705 Correct number of masked data values!')
+    df_points +=2
+else:
+    print('\u274C The amount of null data in your dataframe is incorrect.')
+
+
+# Ensure that the date column is the index
+if isinstance(student_ndvi_df.index, pd.core.indexes.datetimes.DatetimeIndex):
+    print('\u2705 You have the index set to the date column!')
+    df_points +=3
+else:
+    print('\u274C You do not have the index set to the date column.')
+
+# Ensure that the date column is datetime
+if isinstance(student_ndvi_df.index[0], pd._libs.tslibs.timestamps.Timestamp):
+    print('\u2705 The data in your date column is datetime!')
+    df_points +=3
+else:
+    print('\u274C The data in your date column is not datetime.')
+
+# Output for timer, # DO NOT MODIFY
+end_time = datetime.now()
+total_time = end_time - start_time
+print(
+    "Your total run time for processing the data was {0}.".format(total_time))
+
+print("\n \u27A1 You received {} out of 10 points for creating a dataframe.".format(
+    df_points))
+
+df_points
+
+
+# In[10]:
+
+
+# Add only the plot code to this cell
+
+# This is the final figure of mean NDVI
+# for both sites across the year
+# with data cleaned to deal with clouds
+
+fig, ax = plt.subplots(figsize = (12, 7))
+
+
+for site, df in final_NDVI_df.dropna().groupby("site"):
+    if site in ["HARV"]:
```

Comment:
@LManak I quite like the if/else labeling you created in your plotting code! Another way to do this with one line would be to use groupby("site"), and then it will just plot the legend for you.